synapse/storage/stream.py

```diff
@@ -586,27 +616,29 @@ def _get_events_around_txn(self, txn, room_id, event_id, before_limit, after_lim
                 "event_id": event_id,
                 "room_id": room_id,
             },
-            retcols=["stream_ordering", "topological_ordering"],
+            retcols=["stream_ordering", "topological_ordering", "chunk_id"],
         )
 
         # Paginating backwards includes the event at the token, but paginating
         # forward doesn't.
         before_token = RoomStreamToken(
-            results["topological_ordering"] - 1,
-            results["stream_ordering"],
+            results["chunk_id"],
+            results["topological_ordering"],
```

Comment:
what's with the change to the -1 here?

Reply:
Oh yes, this was a bug in that to get the token before you take one from `stream_ordering` rather than `topological_ordering`. I guess it matters less in this new world as `topological_ordering` is unique.